amlb/frameworks/definitions.py

```diff
@@ -233,3 +239,45 @@ def _remove_frameworks_with_unknown_parent(frameworks: Namespace):
             "Removing framework %s as parent %s doesn't exist.", framework, parent
         )
         del frameworks[framework]
+
+
+@dataclass
+class Image:
+    author: str
+    image: str
+    tag: str
+
+
+@dataclass
+class Framework:
+    name: str
+    abstract: bool
+    module: str
+    version: str
+    # Image
+    image: Image
+    # Setup
+    _setup_cmd: str | None
+    setup_cmd: str | None
+    setup_script: str | None
+    setup_env: dict = field(default_factory=dict)
+    setup_args: list[str] = field(default_factory=list)
+    # more optionals
+    params: dict = field(default_factory=dict)
+    refs: list = field(default_factory=list)
+    description: str | None = None
+    project: str | None = None
+
+    def __post_init__(self):
+        if isinstance(self.image, dict):
+            self.image = Image(**self.image)
```

Comment:
This helps where framework namespaces were initiated based on nested dicts (loaded from configuration files).